### DISCUSSION TITLE
Extract the inferred type is not working as expected

### DISCUSSION THREAD

**Discussion by wesleyfeitosa** [colinhacks/zod#2772] [Q&A] [answered]
In documentation says that we can infer the type from a schema like this:

![image](https://github.com/colinhacks/zod/assets/32942055/f4152ac1-d07d-45a4-834d-04ff9b412ae7)

But when I try this with the example above, I was expecting attributes required by default and optional only for those I set as optional...
```
import { z } from 'zod';

export const createDiscountCouponSchema = z.object({
  accountId: z.number().optional(),
  coupon: z.string(),
  additionalDiscount: z.number(),
  expirationDate: z.date(),
  status: z.string(),
  campaign: z.string().optional(),
});

export type CreateDiscountCouponDto = z.infer<
  typeof createDiscountCouponSchema
>;
```

however, I'm receiving all attributes as optionals, 

![image](https://github.com/colinhacks/zod/assets/32942055/0a3ee60e-26f6-46e2-93bd-f265872bdb0f)

is that right?

I'm using zod ```3.22.2``` and typescript ```5.0.3```

**Comment by antonmoiseev** [comment]
I have the same issue. zod - `3.22.2` (tried with `3.21.4` as well), typescript - `5.1.6`

**Comment by dudubernardino** [comment, marked as answer]
it works for me when i changed strict to true, take a look in docs about that [requirements](https://zod.dev/?id=requirements)

```javascript
// tsconfig.json
{
  // ...
  "compilerOptions": {
    // ...
    "strictNullChecks": true
  }
}
```

**Comment by wesleyfeitosa** [comment]
> it works for me when i changed strict to true, take a look in docs about that [requirements](https://zod.dev/?id=requirements)
> 
> ```js
> // tsconfig.json
> {
>   // ...
>   "compilerOptions": {
>     // ...
>     "strictNullChecks": true
>   }
> }
> ```

Thanks @dudubernardino, it worked for me
